articles/how-to-upload-images-to-cloudinary-using-node.js/index.md

```diff
@@ -0,0 +1,337 @@
+    "main": "index.js",
+    "scripts": {
+        "start": "node index.js",
+        "dev": "nodemon index.js",
+        "test": "echo \"Error: no test specified\" && exit 1"
+    },
+    "repository": {
+        "type": "git",
+        "url": "git+https://github.com/RisoriTofa/Cloudinary-uploader-Node.js.git"
+    },
+    "keywords": [],
+    "author": "",
+    "license": "ISC",
+    "bugs": {
+        "url": "https://github.com/RisoriTofa/Cloudinary-uploader-Node.js/issues"
+    },
+    "homepage": "https://github.com/RisoriTofa/Cloudinary-uploader-Node.js#readme",
+    "dependencies": {
+        "cloudinary": "^1.25.1",
+        "dotenv": "^8.2.0",
+        "express": "^4.17.1",
+        "formidable": "^1.2.2",
+        "nodemon": "^2.0.7"
+    }
+}
+```
+
+### Setting up our starting point
+
+Inside the "index.js" file, we shall do the following in a sequential process:
+
+- Requiring modules for this project. This is shown in the code below:
+
+```javascript
+// Required modules
+const http = require('http');
+const util = require('util');
+
+// https://github.com/node-formidable/node-formidable
+const Formidable = require('formidable');
+
+//https://www.npmjs.com/package/dotenv
+const cloudinary = require("cloudinary");
+require('dotenv').config()
+```
+
+- Setting up our Cloudinary configuration. In it, we shall set three parameters that are stored in the ".env" file. These include the Cloud name (CLOUD_NAME), our API key (API_KEY), and lastly the API secret (API_SECRET). This is shown in the code below:
+
+```javascript
+// Cloudinary configuration settings
+// This will be fetched from the .env file in the root directory
+cloudinary.config({
+    cloud_name: process.env.CLOUD_NAME,
+    api_key: process.env.API_KEY,
+    api_secret: process.env.API_SECRET
+});
+```
+
+- We shall create a simple Node.js server to run our project in it. In this server, we shall call on a cloudinary function named "upload" to fetch the image in the path selected and to upload it into our cloudinary account. We shall then display the results both in the terminal and in the browser as plain text when the request inside our URL is `/upload` and a `post` method is used. This is the uploaded file's metadata. This is shown below:
+
+```javascript
+//Create a server
+http.createServer((req, res) => {
+    if (req.url === '/upload' && req.method.toLowerCase() === 'post') {
+
+        // parse a file upload
+        const form = new Formidable();
+
+        form.parse(req, (err, fields, files) => {
+
+            // Find Cloudinary documentation using the link below
+            // https://cloudinary.com/documentation/upload_images
+            cloudinary.uploader.upload(files.upload.path, result => {
+
+                console.log(result)
+                if (result.public_id) {
+                    res.writeHead(200, { 'content-type': 'text/plain' });

```

Comment:
Please include more code comments that give new readers a good overview on the code snippets

Reply:
Done